pkg/operator/status.go

```diff
@@ -117,6 +116,7 @@ func (optr *Operator) syncAvailableStatus() error {
 		}
 
 		optr.eventRecorder.Eventf(mcoObjectRef, corev1.EventTypeWarning, "OperatorNotAvailable", message)
+		message = fmt.Sprintf("Cluster not available for %s", co.Status.Versions)
```

Comment:
Am I correct in understanding that, theoretically, this failure can happen in one of two cases:

1. the MCO is not performing an upgrade, but e.g. the user touched the daemonset definition and now it is having trouble rendering MCD definitions. We should report failing on current version
2. we are performing an upgrade, and the new incoming MCO is failing to sync on something (this is far more likely). 

In the second case, should we maybe distinguish that fact here somehow? i.e. technically the new version is being deployed, and we can no longer roll back to the old version anymore. But the new version is failing right now, not the old one.

Reply:
This is just moving the same message we'd always log (if unavailable)  just moved down (from line 111) and logged less (not the reqd pools case). Mostly I'd like to merge this to to get some signal from CI and potentially iterate.

Reply:
Took a quick look at recommend way by CVO on how it should report different state of operator. Overall this looks fine, we will probbaly need to fix messaging according to doc https://github.com/openshift/enhancements/blob/master/dev-guide/cluster-version-operator/dev/clusteroperator.md#conditions . What I understood from doc, if there is an error, then operator should report following:
```
If an error blocks reaching 4.0.1, the conditions might be:

    Degraded is true with a detailed message Unable to apply 4.0.1: could not update 0000_70_network_deployment.yaml because the resource type NetworkConfig has not been installed on the server.
    Available is true with message Cluster has deployed 4.0.0
    Progressing is true with message Unable to apply 4.0.1: a required object is missing
```

This means if we are reporting available it would report existing version from which upgrade started.

Does `co.Status.Versions` target updated version or the existing version from which upgrade started?

Reply:
Yeah I actually fixed that in the other commit: https://github.com/openshift/machine-config-operator/pull/2721/commits/28c305d9550946702f145ba061d6cec8fcf9502f

co.status.version is the actual version the MCO is on, not the incoming one

Reply:
I noticed we were doing it wrong before, but will also verify things once I get some periodic runs in. :smile_cat: